src/backend/marsha/core/tests/test_api_document.py

```diff
@@ -0,0 +1,371 @@
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def test_api_document_update_instructor(self):
+        """An instructor should be able to update a video."""
+        document = DocumentFactory()
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(document.id)
+        jwt_token.payload["roles"] = ["instructor"]
+        jwt_token.payload["read_only"] = False
+        data = {"title": "new title"}
+
+        response = self.client.put(
+            "/api/documents/{!s}/".format(document.id),
+            json.dumps(data),
+            HTTP_AUTHORIZATION="Bearer {!s}".format(jwt_token),
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+
+        document.refresh_from_db()
+        self.assertEqual(document.title, "new title")
+
+    def test_api_document_initiate_upload_anonymous(self):
+        """Anonymous user should not be able to initiate an upload."""
+        document = DocumentFactory()
+        response = self.client.post(
+            "/api/documents/{!s}/initiate-upload/".format(document.id)
+        )
+        self.assertEqual(response.status_code, 401)
+
+    def test_api_document_initiate_upload_student(self):
+        """Student should not be able to initiate an upload."""
+        document = DocumentFactory()
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(document.id)
+        jwt_token.payload["roles"] = ["student"]
+        jwt_token.payload["read_only"] = False
+
+        response = self.client.post(
+            "/api/documents/{!s}/initiate-upload/".format(document.id),
+            HTTP_AUTHORIZATION="Bearer {!s}".format(jwt_token),
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def test_api_document_initiate_upload_instructor_read_only(self):
+        """An instructor should not be able to initiate an upload for a document in read_only."""
+        document = DocumentFactory()
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(document.id)
+        jwt_token.payload["roles"] = ["instructor"]
+        jwt_token.payload["read_only"] = True
+
+        response = self.client.post(
+            "/api/documents/{!s}/initiate-upload/".format(document.id),
+            HTTP_AUTHORIZATION="Bearer {!s}".format(jwt_token),
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def test_api_document_initiate_upload_instructor(self):
+        """An instructor should not be able to initiate an upload for a document."""
+        document = DocumentFactory(
+            id="27a23f52-3379-46a2-94fa-697b59cfe3c7",
+            upload_state=random.choice(["ready", "error"]),
+        )
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(document.id)
+        jwt_token.payload["roles"] = ["instructor"]
```

Comment:
maybe on all these tests you could replace:
```suggestion
        jwt_token.payload["roles"] = [random.choice(["instructor", "admin"])]
```
so that we test the admin role as well... what do you think?

Reply:
:+1: